back/engines/commercial/content_builder/app/models/content_builder/layout.rb

```diff
@@ -74,11 +74,15 @@ def sanitize_craftjs_json
     end
 
     def set_craftjs_json
-      return if code != 'homepage' || craftjs_json.present?
+      return if code != 'homepage'
 
-      craftjs_filepath = Rails.root.join('config/homepage/default_craftjs.json.erb')
-      json_craftjs_str = ERB.new(File.read(craftjs_filepath)).result(binding)
-      self.craftjs_json = ContentBuilder::LayoutImageService.new.swap_data_images(JSON.parse(json_craftjs_str))
+      if craftjs_json.present?
+        self.craftjs_json = ContentBuilder::LayoutImageService.new.swap_data_images(craftjs_json)
```

Comment:
Why do we suddenly have to 'swap data images' here, while we did not before?

Reply:
We did have it before.

Before, a new homepage would have no `craftjs_json`, and so we would get it from `default_craft.json.erb` and then run `swap_data_images` on the new craftjs_json.
Now, when the `craftjs_json` is set as the homepage `layout` record is created from the internal template we don't need to get the craftjs_json from the erb file, but we still need to run `swap_data_images` on it.

Reply:
I've added a new hook `before_validation :swap_data_images, on: :create`, so now we only run `swap_data_images` once when a new homepage layout is created (usually when a new tenant is created), and not every validation.